code/modules/status_system/wraithEffects.dm

```diff
@@ -0,0 +1,260 @@
+					if (4)
+						random_brute_damage(H, 3)
+						H.playsound_local(H, "sound/impact_sounds/Flesh_Tear_[pick("1", "2", "3")].ogg", 70)
+						boutput(H, pick("<span class='alert'>SOMETHING BIT YOU, HOLY SHIT!!!</span>"))
+		if ((duration >= 90 SECONDS) && (duration < 130 SECONDS))
+			SPAWN(5 DECI SECONDS)
+				H.playsound_local(H, "sound/effects/heartbeat.ogg", 70)
+			H.make_jittery(30)
+			H.setStatus("terror", 30 SECONDS)
+			if(icon_state != "dread4")
+				icon_state = "dread4"
+			if(probmult(15))
+				switch (rand(1, 4))
+					if (1)
+						H.take_brain_damage(10)
+						boutput(H, pick("<span class='alert'>YOU CANT TAKE IT ANYMORE!!!</span>", "<span class='alert'>This cant be real!</span>", "<span class='alert'>It's ALL LIES! ALL OF IT!</span>"))
+					if (2)
+						H.emote("scream")
+						boutput(H, "<span class='alert'>NO, NO, NO!</span>")
+					if (3)
+						H.emote("panic")
+						boutput(H, pick("<span class='alert'>IT'S RIGHT HERE, YOU JUST CAN'T SEE IT!</span>", "<span class='alert'>IT'S WATCHING YOU, LAUGHING! YOU KNOW IT!</span>"))
+					if (4)
+						H.contract_disease(/datum/ailment/malady/heartfailure,null,null,1)	//Bad luck
+						H.visible_message("<span class='alert'>[H] suddenly clutches their chest with a terrified expression</span>", "<span class='alert'>Your heart is beating out of your chest! You feel like death!</span>")
+		if ((duration >= 130 SECONDS))
+			H.contract_disease(/datum/ailment/malady/heartfailure,null,null,1)
+			H.contract_disease(/datum/ailment/malady/flatline,null,null,1)
+			H.visible_message("<span class='alert'>[H]'s face goes blank as they start to collapse to the ground</span>", "<span class='alert'>Your nerves can't take it any longer! Your heart is giving up on you!</span>")
+			duration = 60 SECONDS
+
+		for (var/obj/item/device/pda2/P in range(3, H)) //Turn off all nearby pda lights
+			if((P.module != null) && istype(P.module, /obj/item/device/pda_module/flashlight))
+				var/obj/item/device/pda_module/flashlight/F = P.module
+				if (F.on)
+					F.toggle_light()
+
+/datum/statusEffect/terror
+	id = "terror"
+	name = "Terror"
+	desc = "terrorized"
+	unique = 1
+	duration = 40 SECONDS
+	maxDuration = 3 MINUTES
+	visible = FALSE
+	var/mob/living/carbon/human/H
+	var/list/wall_list = list()
+	var/sound_effect = null
+	var/illusion_icon = null
+	var/illusion_icon_state = null
+	var/volume = null
+	var/has_faked_armory = false
+	var/has_faked_nuke = false
+	var/has_faked_shuttle = false
+
+	onAdd(optional=null)
+		. = ..()
+		if (ishuman(owner))
+			H = owner
+			get_image_group(CLIENT_IMAGE_GROUP_ILLUSSION).add_mob(H)
+		else
+			owner.delStatus("terror")
+
+	onRemove()
+		. = ..()
+		H = owner
+		get_image_group(CLIENT_IMAGE_GROUP_ILLUSSION).remove_mob(H)
+
+	onUpdate()
+		var/mult = 1
+		if (probmult(5))
+			switch (rand(1,3))
+				if (1) // Image based illusion
+					for (var/turf/W in range(6, H))	//Check for surrounding spots
+						wall_list += W
```

Comment:
should be a block(), and isn't even checking for walls lol, so much code like this